src/code/opsButton.js

```diff
@@ -0,0 +1,292 @@
+    this._displayDialog();
+  },
+
+  _displayDialog: function() {
+    var op = window.plugin.Wasabee.getSelectedOperation();
+    var content = document.createElement("div");
+    content.id = "wasabee-dialog-operation-content";
+    var opinfo = document.createElement("div");
+    opinfo.id = "wasabee-dialog-operation-opinfo";
+
+    content.appendChild(this._opSelectMenu(this, op));
+    content.appendChild(opinfo);
+
+    var obHandler = this;
+    this._dialog = window.dialog({
+      title: "Operations",
+      width: "auto",
+      height: "auto",
+      html: content,
+      dialogClass: "wasabee-dialog-mustauth",
+      closeCallback: function() {
+        obHandler.disable();
+        delete obHandler._dialog;
+      },
+      id: window.plugin.Wasabee.static.dialogNames.opsButton
+    });
+    let operationSelect = document.getElementById("wasabee-operationSelect");
+    $(operationSelect).change();
+  },
+
+  removeHooks: function() {
+    Feature.prototype.removeHooks.call(this);
+  },
+
+  _closeDialogs: function() {
+    Object.values(window.plugin.Wasabee.static.dialogNames).forEach(function(
+      name
+    ) {
+      if (name != window.plugin.Wasabee.static.dialogNames.opsButton) {
+        let id = "dialog-" + name;
+        if (window.DIALOGS[id]) {
+          try {
+            let selector = $(window.DIALOGS[id]);
+            selector.dialog("close");
+            selector.remove();
+          } catch (err) {
+            console.log("closing dialog: " + err);
+          }
+        }
+      }
+    });
+  },
+
+  _opSelectMenuUpdate: function(context, operation) {
+    var dialogcontainer = document.getElementById(
+      "wasabee-dialog-operation-content"
+    );
+    var oldspinner = document.getElementById(
+      "wasabee-dialog-operation-spinner"
+    );
+    dialogcontainer.replaceChild(
+      context._opSelectMenu(context, operation),
+      oldspinner
+    );
+  },
+
+  _opSelectMenu: function(context, operation) {
+    var container = document.createElement("div");
+    container.id = "wasabee-dialog-operation-spinner";
+    container.className = "spinner";
+    container.innerHTML = "";
+    $(container).css({
+      "text-align": "center"
+    });
+    var operationSelect = document.createElement("select");
+    operationSelect.id = "wasabee-operationSelect";
+    $(operationSelect).css({
+      width: "50%"
+    });
+    var ol = window.plugin.wasabee.opsList();
+    ol.forEach(function(opID) {
+      var op = window.plugin.wasabee.getOperationByID(opID);
+      $(operationSelect).append(
+        $("<option>").prop({
+          value: opID,
+          text: op.name
+        })
+      );
+    });
+    $(operationSelect).val(operation.ID);
+    $(operationSelect).change(function() {
+      var newID = $(this).val();
+      console.log("load requested for " + newID);
+      // close all other dialogs or things get weird
+      Object.values(window.plugin.Wasabee.static.dialogNames).forEach(function(
```

Comment:
Here's another place where that close all dialogs utility function would be used.